src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ConversionCompletionProviderTests.cs

```diff
@@ -0,0 +1,1349 @@
+            await VerifyCustomCommitProviderAsync(@"
+public struct S {
+    public static explicit operator int(S _) => 0;
+}
+public class Program
+{
+    public static void Main()
+    {
+        S? s = null;
+        s.$$
+    }
+}
+", "int", @"
+public struct S {
+    public static explicit operator int(S _) => 0;
+}
+public class Program
+{
+    public static void Main()
+    {
+        S? s = null;
+        ((int?)s).$$
+    }
+}
+");
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(47511, "https://github.com/dotnet/roslyn/issues/47511")]
+        public async Task ExplicitConversionOfConditionalAccessOfStructAppliesNullableStruct()
+        {
+            // see https://sharplab.io/#gist:08c697b6b9b6384b8ec81cc586e064e6 to run a sample
+            // conversion ((int)c?.S) fails with System.InvalidOperationException: Nullable object must have a value.
+            // conversion ((int?)c?.S) passes (returns an int? with HasValue == false)
+            await VerifyCustomCommitProviderAsync(@"
+public struct S {
+    public static explicit operator int(S _) => 0;
+}
+public class C {
+    public S S { get; } = default;
+}
+public class Program
+{
+    public static void Main()
+    {
+        C c = null;
+        c?.S.$$
+    }
+}
+", "int", @"
+public struct S {
+    public static explicit operator int(S _) => 0;
+}
+public class C {
+    public S S { get; } = default;
+}
+public class Program
+{
+    public static void Main()
+    {
+        C c = null;
+        ((int?)c?.S).$$
+    }
+}
+");
+        }
+
+        [WpfTheory, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(47511, "https://github.com/dotnet/roslyn/issues/47511")]
+        // From property is not Nullable<From>
+        [InlineData("class", "class", false, false, false)] // c.From class -> class = (To)
+        [InlineData("class", "class", false, true, false)] // c?.From class -> class = (To)
+        [InlineData("struct", "class", false, false, false)] // c.From struct -> class = (To)
+        [InlineData("struct", "class", false, true, false)] // c?.From struct -> class = (To)
+        [InlineData("class", "struct", false, false, false)] // c.From class -> struct = (To)
+        [InlineData("class", "struct", false, true, false)] // c?.From class -> struct = (To)
+        [InlineData("struct", "struct", false, false, false)] // c.From struct -> struct = (To)
+        [InlineData("struct", "struct", false, true, true)] // c?.From struct -> struct = (To?)
+
+        // From property is Nullable<From>
+        [InlineData("struct", "class", true, false, false)] // c.From struct -> class = (To)
+        [InlineData("struct", "class", true, true, false)] // c?.From struct -> class = (To)
+        [InlineData("struct", "struct", true, false, true)] // c.From struct -> struct = (To?)
+        [InlineData("struct", "struct", true, true, true)] // c?.From struct -> struct = (To?)
```

Comment:
i think a cleaner way here would be to have an enums that varies between struct/class.  then, we can do a combinatoric test and xuit will try all these cases.

Reply:
Done in 0f29687 (but without using enums, I don't think the enums would improve readability and maintainability is not much of a concern here). I don't like the fact, that we need to ignore some combinations (NRT cases should not be tested here IMO). On the other hand, I like how `assertShouldBeNullable` reads. 

https://github.com/dotnet/roslyn/blob/0f29687bc1a547c4e7fedc1d338b20cbc9b0481b/src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ConversionCompletionProviderTests.cs#L929-L954